File: internal/inventory/inventory.go
package inventory

import (
	"context"

	"github.com/ryanadiputraa/inventra/domain"
)

type ItemPayload struct {
	ItemName string         `json:"item_name" validate:"required"`
	Type     string         `json:"type" validate:"required"`
	Prices   []PricePayload `json:"prices" validate:"required,min=1,dive"`
}

type PricePayload struct {
	Price    int `json:"price" validate:"required"`
	Quantity int `json:"quantity" validate:"required,min=1"`
}

type InventoryService interface {
	AddItem(ctx context.Context, organizationID int, payload ItemPayload) (domain.Item, error)
	ListItems(ctx context.Context, organizationID, page, size int) ([]domain.Item, int64, error)
}

type InventoryRepository interface {
	SaveItem(ctx context.Context, item domain.Item, prices []domain.ItemPrice) (domain.Item, error)
	FetchItems(ctx context.Context, organizationID, page, size int) ([]domain.Item, int64, error)
}

File: frontend/src/types/inventory.ts
export type Item = {
  id: number;
  item_name: string;
  item_type: ItemType;
  stock: ItemPrice[];
  created_at: string;
};

export type ItemType = "consumable" | "fixed_asset";

export type ItemPrice = {
  price: number;
  quantity: number;
  created_at: string;
};

export type AddItemPayload = {
  item_name: string;
  type: ItemType;
  prices: ItemPricePayload[];
};

export type ItemPricePayload = {
  price: number;
  quantity: number;
};
